presto-oracle/src/main/java/io/prestosql/plugin/oracle/OracleClient.java

```diff
@@ -102,6 +119,25 @@ public OracleClient(
         return new String[] {"TABLE", "VIEW"};
     }
 
+    @Override
+    protected Collection<String> listSchemas(Connection connection)
+    {
+        try (ResultSet resultSet = connection.getMetaData().getSchemas(connection.getCatalog(), null)) {
+            ImmutableSet.Builder<String> schemaNames = ImmutableSet.builder();
+            while (resultSet.next()) {
+                String schemaName = resultSet.getString("TABLE_SCHEM");
+                // skip internal schemas
+                if (!INTERNAL_SCHEMAS.contains(schemaName.toLowerCase(ENGLISH))) {
```

Comment:
Introduce a `filterSchema` method in BaseJdbcClient, so that you need to override only that, instead of whole `listSchemas`.

Use this new method in `io.prestosql.plugin.jdbc.BaseJdbcClient#getTableNames` too.
(This could solve https://github.com/prestosql/presto/issues/3831 as well)

see `io.prestosql.plugin.hive.HiveMetadata#filterSchema`

Reply:
Let me separate PR as #3834